packages/flow-core/src/components/f-document-viewer/f-document-viewer.ts

```diff
@@ -0,0 +1,197 @@
+import { HTMLTemplateResult, PropertyValueMap, unsafeCSS } from "lit";
+import { property, query } from "lit/decorators.js";
+import eleStyle from "./f-document-viewer.scss?inline";
+import globalStyle from "./f-document-viewer-global.scss?inline";
+import { FRoot } from "../../mixins/components/f-root/f-root";
+import { FDiv } from "../f-div/f-div";
+import { flowElement } from "../../utils";
+import { injectCss } from "@cldcvr/flow-core-config";
+import { FText } from "../f-text/f-text";
+import { FAccordion } from "../f-accordion/f-accordion";
+import { FSearch } from "../f-search/f-search";
+import { FIconButton } from "../f-icon-button/f-icon-button";
+import { FRadio } from "../f-radio/f-radio";
+import { FDivider } from "../f-divider/f-divider";
+import { FPopover } from "../f-popover/f-popover";
+import { FSpacer } from "../f-spacer/f-spacer";
+import { FIcon } from "../f-icon/f-icon";
+import render, { contentSectionHeader, jumplinksItems, jumplinksSection } from "./render";
+import {
+	checkInView,
+	filterObjectByLevel,
+	getAccordionPanel,
+	getTextPanel,
+	handleSearch,
+	handleSelection,
+	scrollToSectionMethod,
+	toggleLeftColumn,
+	traverse
+} from "./handler";
+import { FSelect } from "../f-select/f-select";
+
+injectCss("f-document-viewer", globalStyle);
+
+export type FDocumentData = Record<string, string | FDocumentStatement>;
+export type FDocTextType = "heading" | "para";
+export type FDocumentStatement = {
+	title: string;
+	data?: FDocumentData;
+	type?: FDocTextType;
+	open: boolean;
+	template?: (highlightText?: string | null) => HTMLTemplateResult;
+};
+
+export type FDocViewerContent = Record<string, FDocumentStatement | string>;
+
+@flowElement("f-document-viewer")
+export class FDocumentViewer extends FRoot {
+	/**
+	 * css loaded from scss file
+	 */
+	static styles = [
+		unsafeCSS(eleStyle),
+		unsafeCSS(globalStyle),
+		...FDiv.styles,
+		...FText.styles,
+		...FAccordion.styles,
+		...FSearch.styles,
+		...FIconButton.styles,
+		...FRadio.styles,
+		...FDivider.styles,
+		...FPopover.styles,
+		...FSpacer.styles
+	];
+
+	/**
+	 * @attribute json object for doc-previewer
+	 */
+	@property({ reflect: false, type: Object })
+	content!: FDocViewerContent;
+
```

Comment:
add the following snipper, this tells that content is required field (useful in types generation)

`/**
	 * mention required fields here for generating vue types
	 */
	readonly required = ["content"];`